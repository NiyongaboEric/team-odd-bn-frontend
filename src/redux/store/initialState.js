const state = {
  auth: {
    signup: {
      spinnerStatus: false,
      signupInputData: {},
      signupData: { },
      signupError: { },
    },
    signin: {
      spinnerStatus: false,
      signinInputData: {},
      signinData: {},
      signinError: {},
    },
  },
  trips: {
    requests: {
      requestsData: [],
      singleRequestData: {},
      requestCommentsData: {},
      requestsError: {},
    },
  },
  viewProfile: {
    profile: {},
  },
};
export default state;
