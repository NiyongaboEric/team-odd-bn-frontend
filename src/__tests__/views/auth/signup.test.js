import React from 'react';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { BrowserRouter as Router } from 'react-router-dom';
import SignUp from '../../../containers/user/signup/signup';
import initialState from '../../../redux/store/initialState';
import {
  redirectInitialState,
  spinnerInitialState,
} from '../../../__mocks__/auth/authMock';

const mockStore = configureStore([]);
let wrapper;
let store;

describe('Test Signup page ', () => {
  it('Test redirect', () => {
    store = mockStore(redirectInitialState);

    wrapper = mount(
      <Router>
        <SignUp
          store={store}
          handleSubmit={async () => {
            jest.fn();
            return jest.fn();
          }}
        />
      </Router>,
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('Router').prop('history').location.pathname).toEqual(
      '/verify-email',
    );
    expect(wrapper.find('Redirect').length).toEqual(1);
  });
  it('Test spinner action', () => {
    store = mockStore(spinnerInitialState);

    wrapper = mount(
      <Router>
        <SignUp
          store={store}
          handleSubmit={async () => {
            jest.fn();
            return jest.fn();
          }}
        />
      </Router>,
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('Spinner').length).toEqual(1);
  });
  it('Test snapshot using the initial state', () => {
    store = mockStore(initialState);

    wrapper = mount(
      <Router>
        <SignUp
          store={store}
          handleSubmit={async () => {
            jest.fn();
            return jest.fn();
          }}
        />
      </Router>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
