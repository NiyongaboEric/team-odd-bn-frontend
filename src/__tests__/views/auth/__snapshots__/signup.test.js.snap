// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Signup page  Test redirect 1`] = `ReactWrapper {}`;

exports[`Test Signup page  Test snapshot using the initial state 1`] = `ReactWrapper {}`;

exports[`Test Signup page  Test spinner action 1`] = `ReactWrapper {}`;
